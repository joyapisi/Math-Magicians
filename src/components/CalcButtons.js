import React from 'react';
import PropTypes from 'prop-types';

const CalcButtons = (props) => {
  const { clickBtn } = props;
  return (
    <section className="calc-outer-container">
      <div className="button-box">
        <div className="calc-keys">
          <button type="button" onClick={() => clickBtn('AC')}>AC</button>
          <button type="button" onClick={() => clickBtn('+/-')}>+/-</button>
          <button type="button" onClick={() => clickBtn('%')}>%</button>
        </div>
        <div className="calc-keys">
          <button type="button" onClick={() => clickBtn('7')}>7</button>
          <button type="button" onClick={() => clickBtn('8')}>8</button>
          <button type="button" onClick={() => clickBtn('9')}>9</button>
        </div>
        <div className="calc-keys">
          <button type="button" onClick={() => clickBtn('4')}>4</button>
          <button type="button" onClick={() => clickBtn('5')}>5</button>
          <button type="button" onClick={() => clickBtn('6')}>6</button>
        </div>
        <div className="calc-keys">
          <button type="button" onClick={() => clickBtn('1')}>1</button>
          <button type="button" onClick={() => clickBtn('2')}>2</button>
          <button type="button" onClick={() => clickBtn('3')}>3</button>
        </div>
        <div className="calc-keys">
          <button type="button" onClick={() => clickBtn('0')}>0</button>
          <button type="button" onClick={() => clickBtn('.')}>.</button>
        </div>
      </div>
      <div className="calc-symbols">
        <button type="button" onClick={() => clickBtn('รท')}>รท</button>
        <button type="button" onClick={() => clickBtn('x')}>ร</button>
        <button type="button" onClick={() => clickBtn('-')}>-</button>
        <button type="button" onClick={() => clickBtn('+')}>+</button>
        <button type="button" onClick={() => clickBtn('=')}>=</button>
      </div>
    </section>
  );
};


export default CalcButtons;
