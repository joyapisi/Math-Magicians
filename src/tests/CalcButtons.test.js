import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import CalcButtons from '../components/CalcButtons';

describe('CalcButtons', () => {
  it('should match snapshot', () => {
    const { container } = render(<CalcButtons clickBtn={() => {}} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should call clickBtn prop when a button is clicked', () => {
    const mockClickBtn = jest.fn();
    const { getByText } = render(<CalcButtons clickBtn={mockClickBtn} />);
    fireEvent.click(getByText('5'));
    expect(mockClickBtn).toHaveBeenCalledWith('5');
  });
});
