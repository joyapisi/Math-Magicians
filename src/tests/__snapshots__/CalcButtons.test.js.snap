// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalcButtons should match snapshot 1`] = `
<div
  class="calc-outer-container"
>
  <button
    type="button"
  >
    AC
  </button>
  <button
    type="button"
  >
    +/-
  </button>
  <button
    type="button"
  >
    %
  </button>
  <button
    class="coloured"
    type="button"
  >
    รท
  </button>
  <button
    type="button"
  >
    7
  </button>
  <button
    type="button"
  >
    8
  </button>
  <button
    type="button"
  >
    9
  </button>
  <button
    class="coloured"
    type="button"
  >
    ร
  </button>
  <button
    type="button"
  >
    4
  </button>
  <button
    type="button"
  >
    5
  </button>
  <button
    type="button"
  >
    6
  </button>
  <button
    class="coloured"
    type="button"
  >
    -
  </button>
  <button
    type="button"
  >
    1
  </button>
  <button
    type="button"
  >
    2
  </button>
  <button
    type="button"
  >
    3
  </button>
  <button
    class="coloured"
    type="button"
  >
    +
  </button>
  <button
    class="spanner"
    type="button"
  >
    0
  </button>
  <button
    type="button"
  >
    .
  </button>
  <button
    class="coloured"
    type="button"
  >
    =
  </button>
</div>
`;
